### PR TITLE
API - getEnabledFeatures

### PR DESCRIPTION
**Pull request by oakbani**

### CODE REVIEW

File: src/Optimizely/Optimizely.php [hunk 1]
```diff
@@ -501,6 +501,32 @@ public function isFeatureEnabled($featureFlagKey, $userId, $attributes = null)
...
 
+    /**
+     * Get keys of all feature flags which are enabled for the user
+     * @param  string User ID
+     * @param  array Associative array of user attributes
+     * @return array List of feature flag keys
+     */
+    public function getEnabledFeatures($userId, $attributes = null)
+    {
+        $enabled_feature_keys = [];
```
Comment (mikeng13) [inline src/Optimizely/Optimizely.php]: Please change all variables to camel case: `$enabledFeatureKeys`

File: tests/OptimizelyTest.php [hunk 2]
```diff
@@ -2159,6 +2159,59 @@ public function testIsFeatureEnabledGivenFeatureFlagIsEnabledAndUserIsNotBeingEx
...
+
+       // Mock isFeatureEnabled to return false for all calls
+        $optimizelyMock->expects($this->exactly(8))
+            ->method('isFeatureEnabled')
+            ->will($this->returnValue(false));
+
+        $this->assertEmpty($optimizelyMock->getEnabledFeatures("user_id", []));
+    }
+
+    public function testGetEnabledFeaturesGivenFeaturesAreEnabledForUser()
```
Comment (mikeng13) [inline tests/OptimizelyTest.php]: I'd suggest adding one more test case where the user also has attributes

**Comment by mikeng13** [comment]
E2E tests pass. Going to merge this in.

File: src/Optimizely/Optimizely.php [hunk 3]
```diff
@@ -501,6 +501,32 @@ public function isFeatureEnabled($featureFlagKey, $userId, $attributes = null)
...
+        if (!$this->_isValid) {
+            $this->_logger->log(Logger::ERROR, "Datafile has invalid format. Failing '".__FUNCTION__."'.");
+            return $enabledFeatureKeys;
+        }
+
+        $featureFlags = $this->_config->getFeatureFlags();
+        foreach ($featureFlags as $feature){
+            $featureKey = $feature->getKey();
+            if ($this->isFeatureEnabled($featureKey, $userId, $attributes) === true) {
+                $enabledFeatureKeys[] = $featureKey;
```
Comment (aliabbasrizvi) [inline src/Optimizely/Optimizely.php]: Wow. Wasn't aware of this syntax. Neat. Just make sure travis passes across all PHP versions.

**Comment by mikeng13** [comment]
E2e tests are once again passing. Merging this PR in.

**Pull request merged**